enhancements/cluster-logging/forward_to_cloudwatch.md

```diff
@@ -0,0 +1,281 @@
+
+>  Log groups define groups of log streams that share the same retention, monitoring, and access control settings ... For example, if you have a separate log stream for the Apache access logs from each host, you could group those log streams into a single log group called MyWebsite.com/Apache/access_log.
+
+In other words a *log stream* corresponds to the smallest distinct source of logs.
+A *log group* is a collection of related *log streams*.
+
+Cluster logging collects logs from two distinct types of log source:
+
+- *container logs* are logs from containers, collected via the container run-time.
+- *node logs* are logs from a cluster node's journald or syslogd, they do *not* come from any container.
+
+**Note**: Log *sources* are different from log *categories*.
+Category *application* includes only container logs.
+Categories *infrastructure* and *audit* include a mix of selected container and node logs.
+
+#### Generating log streams
+
+This proposal generates a unique *log stream* for each distinct container or node log source.
+The *log stream name* is constructed from meta-data including the container/node UUID.
+
+**Note**: The log stream name should be considered *opaque* by the end user.
+It should *not* be used for indexing, searching or as a reliable source of meta-data.
+The end user can retrieve all meta-data as JSON fields in the log record.
+In future we may explore optimizations that change the stream names, by encoding static meta-data in the name.
+Making the names opaque, allows for future change without breaking user assumptions.
+
+#### Generating log groups
+
+The choice of *Log group* names is configured by the user, they are not generated by the forwarder.
+Log group names are identical to some other well-known identifier.
+
+Log groups can be named after:
+
+- **log category**: "application", "infrastructure" and "audit".
+  A single group for each category.
+- **namespace name**: Group per namespace *name*.
+  Used when successive namespace objects with the same name are considered "equivalent".
+  This is a common case, many core k8s tools and APIs work this way.
+- **namespace uuid**: Group per namespace *object*.
+  Destroying then creating a namespace object with the same name results in a *new log group*.
+  Use when it is important to distinguish logs from successive namespaces instances with the same name.
+  For example, when namespace re-creation is considered a security risk.
+
+### API fields
+
+New API fields in the `output.cloudwatch` section:
+
+- `region`: (string) AWS region name, required to connect.
+- `groupBy`: (string, default "category") Take group name from logging meta-data. Values:
+   - `category`: category of log entry - one of "application", "infrastructure", or "audit"
+   - `namespaceName`: Container's namespace name
+   - `namespaceUUID`: Container's namespace UUID
```

Comment:
if there is any error fetching meta info from the API server this will fail? What should the fall back be?  namespace, podname, container hash are available from the container file path and don't rely on an api call

Reply:
No fallback, if the metadata is missing we don't set a group. It will default to whatever AWS does if no group is explicitly set. Note this is the *group* name not the *stream* name.